discovery/db_discovery.go

```diff
@@ -108,9 +108,13 @@ func (dbo *DBOrchestratorPoolCache) GetOrchestrators(numOrchestrators int, suspe
 
 		// check if O's price is below B's max price
 		maxPrice := server.BroadcastCfg.MaxPrice()
-		price := big.NewRat(info.PriceInfo.PricePerUnit, info.PriceInfo.PixelsPerUnit)
+		price, err := common.RatPriceInfo(info.PriceInfo)
```

Comment:
I think problems can still occur if `info.PriceInfo` is nil. In this case, `RatPriceInfo()` will return `nil, nil`. Since the error is nil, a nil `price` will be used in the below conditional which would trigger a nil pointer error. We should make sure that `price` is not nil before using it.